### PR TITLE
xds: move balancer/resolver registation to package xds

### PR DESCRIPTION
**Pull request by menghanl**
Instead of package "xds/experimental".

### CODE REVIEW

File: xds/xds.go [hunk 1]
```diff
@@ -16,10 +16,10 @@
...
  */
 
-// Package experimental contains xds implementation, still in experimental
-// state. Users only need to import this package to get all xds functionality.
-// Things are expected to change fast until we get to a stable state, at
-// which point, all this will be moved to the xds package.
+// Package xds contains xds implementation, still in experimental state. Users
+// only need to import this package to get all xds functionality.  Things are
+// expected to change fast until we get to a stable state, at which point, all
+// this will be moved to the xds package.
```
Comment (dfawley) [inline xds/xds.go:22]: This is stale now, right?

Comment (menghanl) [reply xds/xds.go]: Fresh now...

File: xds/xds.go [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+// Package xds contains xds implementation. Users need to import this package to
```
Comment (dfawley) [inline xds/xds.go:19]: Is there some canonical documentation we can link to about what xDS is and how to get started?

Comment (menghanl) [reply xds/xds.go:19]: Not yet. Our doc doesn't have instruction to setup the control plane. The wallet example would be a good place, but it doesn't exist.

Comment (dfawley) [reply xds/xds.go:19]: Will we be adding something to http://grpc.io?  Let's try to remember to come back and add a pointer when we have one.

Comment (dfawley) [reply xds/xds.go:19]: Maybe a link to envoy for now?  As a user that doesn't know what xds is, this message means nothing to me: "xds contains xds. Import this to get xds"  :smile:

Comment (menghanl) [reply xds/xds.go:19]: I linked to the xds example. At least that (tried to) explain what xDS is. And we would update the example to point to whatever new doc we will have.

Comment (dfawley) [reply xds/xds.go:19]: This is fine for now.  We should definitely have a link to our "official" xds documentation from this package's godoc once it exists.

Comment (menghanl) [reply xds/xds.go:19]: We tried this same thing when writing the xds example, and failed to find a good link to post there... We will need our official doc.

Comment (easwars) [reply xds/xds.go:19]: How about the xDS gRFC?

Comment (menghanl) [reply xds/xds.go:19]: The example readme does link to the gRFC.

**Pull request merged**